tracing/setup_overview/setup: update Go env. vars. and setup for UDS and ports (#13363)
DataDog/documentation: content/en/tracing/setup_overview/setup/go.md

3. After the application is instrumented, the trace client attempts to send traces to the Unix domain socket `/var/run/datadog/apm.socket` by default. If the socket does not exist, traces are sent to `http://localhost:8126`.

   A similar rule applies to all metrics sent by the Go tracer (including Runtime Metrics and internal telemetry): the client will attempt to send Dogstatsd data to the Unix domain socket `/var/run/datadog/dsd.socket` and will default to `http://localhost:8125` if that does not exist.

>    If different hosts or ports are required, one or more of the below environment variables can be used. The examples show the defaults but can be set to other values as well.

Reviewer: ```suggestion
   If you require different hosts or ports, use one or more of the following environment variables. The examples show the defaults, but you can set them to other values as well.
```

   ```

>    DD_AGENT_HOST=localhost   # the default host to send traces and metrics too

Reviewer 2: Since **DD_AGENT_HOST** defaults to localhost when not set, should 

```
DD_AGENT_HOST=localhost   # the default host to send traces and metrics too
```

be something like:

```
DD_AGENT_HOST=localhost   # the host to send traces; defaults to `localhost` when not set
```

?

And should the description note this is for traces only, while metrics follow dogstatsd?

Author: It's the agent host. For both.

Reviewer 2: Okay! I'll resolve this comment.

   DD_TRACE_AGENT_PORT=8126  # the default port to send traces too

>    DD_DOGSTATSD_PORT=8125    # the default port to send Dogstatsd metrics too

Author: Done. Thanks

   ```

   The connection for traces can also be configured in code:

    ```go
    package main

    import "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"

    func main() {
        tracer.Start(
            // Unix Domain Socket configuration:
            tracer.WithUDS("/var/run/datadog/apm.socket"),
            // or, for alternative TCP connection:
            // tracer.WithAgentAddr("localhost:8126"),
            // or, for alternative Dogstatsd connection:

>             // tracer.WithDogstatsdAddress("localhost:8125"),

Author: No. Do you reckon I should mention "UDP" to be clear?

>             // tracer.WithDogstatsdAddress("localhost:8125"),

Reviewer 2: Yes, specifying that this is an `alternative Dogstatsd connection for UDP` would prevent confusion in trying to use it for UDS.

>             // tracer.WithDogstatsdAddress("localhost:8125"),

Author: Done
